src/symtable/SymbolTableScope.ts

```diff
@@ -332,12 +331,7 @@ export class SymbolTableScope {
         }
     }
 
-    updateSymbols(
-        scopes: { [name: string]: number },
-        bound: Set<string> | null,
-        free: Set<string>,
-        classFlag: boolean
-    ) {
+    updateSymbols(scopes: FlagMap, bound: Set<string> | null, free: Set<string>, classFlag: boolean) {
```

Comment:
There's a bug in this function

```typescript
for (const name in free) 
```
should be
```typescript
for (const name of free) 
```

Reply:
I saw! good catch